pixi.toml

```
@@ -228,6 +228,14 @@ rerun-build-web-release = "rustup target add wasm32-unknown-unknown && cargo run
 
 rs-check = "python scripts/ci/rust_checks.py"
 
+# Run basic rust checks: formatting, clippy, and tests.
+# Faster alternative to rs-check-everything, skips lint-rerun and extra checks.
+rs-check-basics = "python scripts/ci/rust_checks.py --only base_checks tests"
+
+# Run all comprehensive checks: linting, tests, clippy, and formatting.
+# This is equivalent to what you should run before pushing to ensure CI will pass.
+rs-check-everything = "python scripts/lint.py && cargo test --quiet && cargo clippy --quiet --all-targets --all-features -- --deny warnings && cargo fmt --all -- --check"
```

```suggestion
rs-check-everything = "python scripts/lint.py && cargo run nextest --quiet --all-targets --all-features && cargo clippy --quiet --all-targets --all-features -- --deny warnings && cargo fmt --all -- --check"
```

is a lot faster and has process isolation, also misses all-targets/all-features flags. That's also what we do on ci.
Not a fan of all these quiet flags personally, but I'm not the target audience anyways ;-)